homeassistant/components/growatt_server/sensor.py

```diff
@@ -345,7 +345,207 @@
+        "Export to grid today",
+        ENERGY_KILO_WATT_HOUR,
+        "etoGridToday",
+        {"device_class": DEVICE_CLASS_ENERGY},
+    ),
+    "mix_export_to_grid_lifetime": (
+        "Lifetime export to grid",
+        ENERGY_KILO_WATT_HOUR,
+        "etogridTotal",
+        {"device_class": DEVICE_CLASS_ENERGY},
+    ),
+    # Values from 'mix_system_status' API call
+    "mix_battery_charge": (
+        "Battery charging",
+        POWER_KILO_WATT,
+        "chargePower",
+        {"device_class": DEVICE_CLASS_POWER},
+    ),
+    "mix_load_consumption": (
+        "Load consumption",
+        POWER_KILO_WATT,
+        "pLocalLoad",
+        {"device_class": DEVICE_CLASS_POWER},
+    ),
+    "mix_wattage_pv_1": (
+        "PV1 Wattage",
+        POWER_WATT,
+        "pPv1",
+        {"device_class": DEVICE_CLASS_POWER},
+    ),
+    "mix_wattage_pv_2": (
+        "PV2 Wattage",
+        POWER_WATT,
+        "pPv2",
+        {"device_class": DEVICE_CLASS_POWER},
+    ),
+    "mix_wattage_pv_all": (
+        "All PV Wattage",
+        POWER_KILO_WATT,
+        "ppv",
+        {"device_class": DEVICE_CLASS_POWER},
+    ),
+    "mix_export_to_grid": (
+        "Export to grid",
+        POWER_KILO_WATT,
+        "pactogrid",
+        {"device_class": DEVICE_CLASS_POWER},
+    ),
+    "mix_import_from_grid": (
+        "Import from grid",
+        POWER_KILO_WATT,
+        "pactouser",
+        {"device_class": DEVICE_CLASS_POWER},
+    ),
+    "mix_battery_discharge_kw": (
+        "Battery discharging kW",
+        POWER_KILO_WATT,
+        "pdisCharge1",
+        {"device_class": DEVICE_CLASS_POWER},
+    ),
+    "mix_grid_voltage": (
+        "Grid voltage",
+        VOLT,
+        "vAc1",
+        {"device_class": DEVICE_CLASS_VOLTAGE},
+    ),
+    # Values from 'mix_detail' API call
+    "mix_system_production_today": (
+        "System production today (self-consumption + export)",
+        ENERGY_KILO_WATT_HOUR,
+        "eCharge",
+        {"device_class": DEVICE_CLASS_ENERGY},
+    ),
+    "mix_load_consumption_solar_today": (
+        "Load consumption today (solar)",
+        ENERGY_KILO_WATT_HOUR,
+        "eChargeToday",
+        {"device_class": DEVICE_CLASS_ENERGY},
+    ),
+    "mix_self_consumption_today": (
+        "Self consumption today (solar + battery)",
+        ENERGY_KILO_WATT_HOUR,
+        "eChargeToday1",
+        {"device_class": DEVICE_CLASS_ENERGY},
+    ),
+    "mix_load_consumption_battery_today": (
+        "Load consumption today (battery)",
+        ENERGY_KILO_WATT_HOUR,
+        "echarge1",
+        {"device_class": DEVICE_CLASS_ENERGY},
+    ),
+    "mix_import_from_grid_today": (
+        "Import from grid today (load)",
+        ENERGY_KILO_WATT_HOUR,
+        "etouser",
+        {"device_class": DEVICE_CLASS_ENERGY},
+    ),
+    # This sensor is manually created using the most recent X-Axis value from the chartData
+    "mix_last_update": (
+        "Last Data Update",
+        "",
```

Comment:
What kind of measurement does this return?

Reply:
It's a time field in HH:MM format, whenever I tried to use the correct timestamp enums though it broke the field.
See forum discussion on it here: https://community.home-assistant.io/t/correct-way-to-represent-a-time-hh-mm-in-a-custom-component/295639

Happy to take an alternative approach if there is a better one.

Reply:
It would be good if we can convert it to a UTC datetime isoformatted string. See our dt util for available tools.

The timestamp device class requires an isoformatted datetime string.

Reply:
Thanks, I'll try this later today when I get some time. I had tried to use the dt utils before but it resulted in only ever seeing an invalid timestamp error in the UI. I don't think I tried using `None` though as my unit of measurement so that may fix it.

Reply:
@MartinHjelmare I think I've got this working now. Is there a way to force the formatting of the timestamp when it is displayed on the lovelace UI?
Currently it always shows as "X Minutes ago" but I'd quite like it to show just as the time e.g. "HH:MM". 
I think this is probably a personal formatting preference, so I shouldn't change it in the integration, but is there a way of changing it in the UI through "preferences" or something like that?

Before I push the change I'm just going to rebuild my docker image to re-test.

Reply:
Here's the formatting options for the entities card in lovelace:
https://www.home-assistant.io/lovelace/entities/#format

Reply:
Thanks again @MartinHjelmare 
It all works nicely.
Before I make a push, I don't know if you're able to help. I'm using the prometheus exporter/plugin for Hass and the timestamp value is always being exported as a value of `0`. 
I can't work out if it's a problem with the prometheus plugin specifically related to timestamps or whether I need to configure it in some way.
If you don't know, or don't have any documentation to point me towards, I'll go and post in the forums. 
I don't want to push up the latest change in case I need to make a follow up one immediately afterwards to resolve this issue.

Reply:
I've done a bit more digging on this one and I think it's an issue with the Prometheus exporter plugin. I've created a forum post for it here: https://community.home-assistant.io/t/prometheus-exporter-non-numeric-values/298356

I'm just testing whether a field of `TIME_SECONDS` instead of `None` will allow for it to be exported correctly by the Prometheus plugin as well as displayed correctly in the Lovelace UI (that way everyone is happy).

As soon as I've confirmed it I'll push the change.

Reply:
We don't allow setting unit of measurement to anything but `None` for timestamp entities.

Reply:
Understood. Turns out it didn't fix my problem anyway (I've done a lot more digging around the prometheus exporter and completely understand why it behaves the way it does).
Latest changes pushed for the timestamp field as discussed.